draft-ietf-httpbis-http2bis.xml

```diff
@@ -1357,20 +1360,27 @@
           DATA frames MAY also contain padding.  Padding can be added to DATA frames to obscure the
           size of messages.  Padding is a security feature; see <xref target="padding"/>.
         </t>
-        <figure anchor="DATAFramePayload">
-          <name>DATA Frame Payload</name>
+        <figure anchor="DATAFrameFormat">
+          <name>DATA Frame Format</name>
           <artwork type="inline"><![CDATA[
- +---------------+
- |Pad Length? (8)|
- +---------------+-----------------------------------------------+
- |                            Data (*)                         ...
- +---------------------------------------------------------------+
- |                           Padding (*)                       ...
- +---------------------------------------------------------------+
+Data Frame {
+  Length (24),
+  Type (8),
+  END_STREAM Flag (1),
+  Unused Flags (2),
+  PADDED Flag (1),
+  Unused Flags (4),
```

Comment:
I believe these flags (and all the other flags) are defined in the wrong bit order.

This is...not entirely well specified in RFC 9000, but the example given in § 1.3 is:

```
Example Structure {
  One-bit Field (1),
  7-bit Field with Fixed Value (7) = 61,
  Field with Variable-Length Integer (i),
  Arbitrary-Length Field (..),
  Variable-Length Field (8..24),
  Field With Minimum Length (16..),
  Field With Maximum Length (..128),
  [Optional Field (64)],
  Repeated Field (8) ...,
}
```

With the following prose after it:

> When a single-bit field is referenced in prose, the position of that field can be clarified by using the value of the byte that carries the field with the field's value set. For example, the value 0x80 could be used to refer to the single-bit field in the most significant bit of the byte, such as One-bit Field in Figure 1.

This implies that `0x80` refers to the first bit of a byte. `END_STREAM` is not 0x80, it's 0x1. All of the flags sections need to be inverted.

Reply:
The prose uses indexed bits - e.g., for `END_STREAM` it says 'When set, bit 0...'. I had assumed that it was indexing from left, this being network byte order and all, but you could also argue it's indexing from least significant bit (although that's pretty odd IMO).

Nevertheless, this protocol interoperates somehow. 

If I read wireshark correctly, your interpretation is correct:

<img width="415" alt="Screen Shot 2021-07-12 at 3 28 58 pm" src="https://user-images.githubusercontent.com/74384/125235537-f80f9d00-e325-11eb-8e98-d356a2d00e0b.png">

However did we get this far?

Reply:
I'm about to push a commit that should correct that. Please review carefully.

We also might want to reconsider the language we use about ordering...